src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/WorkflowRunRepository.kt

```diff
@@ -17,15 +24,45 @@ class OrderlyWebWorkflowRunRepository : WorkflowRunRepository
     {
         JooqContext().use {
             it.dsl.insertInto(Tables.ORDERLYWEB_WORKFLOW_RUN)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.NAME, workflowRun.name)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.KEY, workflowRun.key)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.EMAIL, workflowRun.email)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.DATE, Timestamp.from(workflowRun.date))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.REPORTS, Gson().toJson(workflowRun.reports))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.INSTANCES, Gson().toJson(workflowRun.instances))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH, workflowRun.gitBranch)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT, workflowRun.gitCommit)
-                .execute()
+                    .set(ORDERLYWEB_WORKFLOW_RUN.NAME, workflowRun.name)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.KEY, workflowRun.key)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.EMAIL, workflowRun.email)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.DATE, Timestamp.from(workflowRun.date))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.REPORTS, Gson().toJson(workflowRun.reports))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.INSTANCES, Gson().toJson(workflowRun.instances))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH, workflowRun.gitBranch)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT, workflowRun.gitCommit)
+                    .execute()
+        }
+    }
+
+    override fun getWorkflowDetails(key: String): WorkflowRun
+    {
+        JooqContext().use {
+            val result = it.dsl.select(
+                    ORDERLYWEB_WORKFLOW_RUN.NAME,
+                    ORDERLYWEB_WORKFLOW_RUN.KEY,
+                    ORDERLYWEB_WORKFLOW_RUN.EMAIL,
+                    ORDERLYWEB_WORKFLOW_RUN.DATE,
+                    ORDERLYWEB_WORKFLOW_RUN.REPORTS,
+                    ORDERLYWEB_WORKFLOW_RUN.INSTANCES,
+                    ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH,
+                    ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT)
+                    .from(ORDERLYWEB_WORKFLOW_RUN)
+                    .where(ORDERLYWEB_WORKFLOW_RUN.KEY.eq(key))
+                    .singleOrNull()
+                    ?: throw UnknownObjectError("key", "workflowDetails")
```

Comment:
```suggestion
                    ?: throw UnknownObjectError("key", "workflow")
```
The second argument should be the name of a type so that the error messages are human-readable e.g. "Unknown workflow : 'incorrect key'"